clap_derive/src/derives/args.rs

```diff
@@ -341,16 +341,10 @@ pub fn gen_augment(
                         #deprecations
 
                         #[allow(deprecated)]
-                        let arg = clap::Arg::new(#id)
-                            #implicit_methods;
-
-                        let arg = arg
-                            #explicit_methods;
-
-                        let arg = arg
-                            #override_methods;
-
-                        arg
+                        clap::Arg::new(#id)
+                            #implicit_methods
+                            #explicit_methods
```

Comment:
This actually changes behavior (expands the scope of the `allow`)

Please instead silence this warning in each derive.  For example, [here](https://github.com/clap-rs/clap/blob/master/clap_derive/src/derives/args.rs#L101) is one place where you would silence it

Reply:
Definitions
- "implicit methods" are ones auto-generated by clap
- "explicit methods" are methods set by the user via attributes
- `#[allow(deprecated)]`  applies to an expression.

Before, `allow(deprecated)` only affected "implicit methods" but now it affects "explicit methods".  As we might not be able to change what we auto-generate for backwards compatibility purposes, we do not want to show users deprecation messages for code we generate.  However, we do want deprecations for code the user writes so they know to update their code.

Reply:
Also, for ease of composing, it can be common in macros to break out expressions into individual statements

Reply:
i see, sorry about that, updated PR to allow `clippy::almost_swapped` in affected codegen paths